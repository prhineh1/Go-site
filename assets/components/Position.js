import React from 'react';
import redChecker from '../imgs/red_checker.svg';
import blackChecker from '../imgs/black_checker.svg';

const Position = (props) => {
    switch(props.content) {
        case 1:
            return <div></div>
        case 'r':
            return <div><img className="piece" src={redChecker} alt="red checker piece" /></div>
        case 'b':
            return <div><img className="piece" src={blackChecker} alt="black checker piece" /></div>
    };
};

export default Position;