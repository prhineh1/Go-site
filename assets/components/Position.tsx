import React, { ReactElement } from 'react';
import PropTypes from 'prop-types';
import redChecker from '../imgs/red_checker.svg';
import blackChecker from '../imgs/black_checker.svg';
import { coordContent } from '../state/actions';
import { Immutable } from '../state/state';

interface PositionProps {
  coord: Immutable<number[]>;
  content: coordContent;
  selected?: (coord: Immutable<number[]>, content: coordContent) => void;
  move?: (toCoords: Immutable<number[]>) => void;
}


const Position: React.FC<PositionProps> = ({
  coord, content, selected, move,
}): ReactElement => (content === 1 ? (
  <button type="button" onKeyDown={(): void => move && move(coord)} onClick={(): void => move && move(coord)} />
) : (
  <button type="button" onClick={(): void => selected && selected(coord, content)} onKeyDown={(): void => selected && selected(coord, content)}>
    <img
      className="piece"
      src={content === 'r' ? redChecker : blackChecker}
      alt={content === 'r' ? 'red checker piece' : 'black checker piece'}
    />
  </button>
));

Position.propTypes = {
  coord: PropTypes.arrayOf(PropTypes.number.isRequired).isRequired,
  content: PropTypes.oneOf<coordContent>([1, 'r', 'b']).isRequired,
  selected: PropTypes.func,
  move: PropTypes.func,
};

Position.defaultProps = {
  selected: (): void => {},
  move: (): void => {},
};

export default Position;
