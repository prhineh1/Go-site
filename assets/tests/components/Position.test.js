import React from 'react';
import Position from '../../components/Position';
import { shallow } from 'enzyme';

test("should render a blank space", () => {
    let wrapper = shallow(<Position coords={[0,0]} content={1} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render a red piece", () => {
    let wrapper = shallow(<Position coords={[0,0]} content={'r'} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render a black piece", () => {
    let wrapper = shallow(<Position coords={[0,0]} content={'b'} />);
    expect(wrapper).toMatchSnapshot();
});