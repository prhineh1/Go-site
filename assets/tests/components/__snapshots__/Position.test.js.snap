// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a black piece 1`] = `
<div>
  <img
    alt="black checker piece"
    className="piece"
    src="test-file-stub"
  />
</div>
`;

exports[`should render a blank space 1`] = `<div />`;

exports[`should render a red piece 1`] = `
<div>
  <img
    alt="red checker piece"
    className="piece"
    src="test-file-stub"
  />
</div>
`;
